From 49408a66a6e766deca0544cb383a06f2cb99d2ca Mon Sep 17 00:00:00 2001
From: Karthik M <quic_karm@quicinc.com>
Date: Sun, 11 Sep 2022 02:20:39 +0530
Subject: [PATCH] ath12k peer pktlog fix

log type and mode of the pktlog need to checked everytime when
skb is dequeued.

peer pktlog is supported only when lite mode is enabled.

Signed-off-by: Karthik M <quic_karm@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/dp_mon.c | 26 +++++++++++-------------
 1 file changed, 12 insertions(+), 14 deletions(-)

diff --git a/drivers/net/wireless/ath/ath12k/dp_mon.c b/drivers/net/wireless/ath/ath12k/dp_mon.c
index 78f894b..9fb033e 100644
--- a/drivers/net/wireless/ath/ath12k/dp_mon.c
+++ b/drivers/net/wireless/ath/ath12k/dp_mon.c
@@ -3759,25 +3759,11 @@ int ath12k_dp_mon_rx_process_stats(struct ath12k *ar, int mac_id,
 	mon_dst_ring = &pdev_dp->rxdma_mon_dst_ring[srng_id];
 	buf_ring = &dp->rxdma_mon_buf_ring;
 
-	if (ath12k_debugfs_is_pktlog_lite_mode_enabled(ar)) {
-		log_type = ATH12K_PKTLOG_TYPE_LITE_RX;
-		rx_buf_sz = DP_RX_BUFFER_SIZE_LITE;
-	} else if (ath12k_debugfs_is_pktlog_rx_stats_enabled(ar)) {
-		log_type = ATH12K_PKTLOG_TYPE_RX_STATBUF;
-		rx_buf_sz = DP_RX_BUFFER_SIZE;
-	}
-
 	srng = &ab->hal.srng_list[mon_dst_ring->ring_id];
 	spin_lock_bh(&srng->lock);
 	ath12k_hal_srng_access_begin(ab, srng);
 
 	while (likely(*budget)) {
-		if (log_type) {
-			trace_ath12k_htt_rxdesc(ar, skb->data, log_type,
-			                        rx_buf_sz);
-			ath12k_rx_stats_buf_pktlog_process(ar, skb->data, log_type,
-			                                   rx_buf_sz);
-		}
 		mon_dst_desc = (struct hal_mon_dest_desc *)
 				ath12k_hal_srng_dst_peek(ab, srng);
 		if (unlikely(!mon_dst_desc))
@@ -3836,6 +3822,18 @@ move_next:
 		ath12k_dp_mon_rx_memset_ppdu_info(ppdu_info);
 
 	while ((skb = __skb_dequeue(&skb_list))) {
+		if (ath12k_debugfs_is_pktlog_lite_mode_enabled(ar)) {
+			log_type = ATH12K_PKTLOG_TYPE_LITE_RX;
+			rx_buf_sz = DP_RX_BUFFER_SIZE_LITE;
+		} else if (ath12k_debugfs_is_pktlog_rx_stats_enabled(ar)) {
+			log_type = ATH12K_PKTLOG_TYPE_RX_STATBUF;
+			rx_buf_sz = DP_RX_BUFFER_SIZE;
+			ath12k_rx_stats_buf_pktlog_process(ar, skb->data, log_type,
+			                                   rx_buf_sz);
+		}
+		if (log_type)
+			trace_ath12k_htt_rxdesc(ar, skb->data, log_type,
+			                        rx_buf_sz);
 
 		hal_status = ath12k_dp_mon_parse_rx_dest(ar, ppdu_info, skb);
 		if (hal_status != HAL_RX_MON_STATUS_PPDU_DONE) {
-- 
2.17.1

